server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1095,7 +1096,13 @@ protected Map<String, String> getConfigs() {
     @Override
     public String generateRandomPassword() {
         final Integer passwordLength = vmPasswordLength.value();
-        return PasswordGenerator.generateRandomPassword(passwordLength);
+        final String passwordComplexity = "^(?=.{6,})(?=.*[A-Z])(?=.*[a-z])(?=.*[@#%&^~.,!,?,:,;])"
```

Comment:
You should use the configuration that you created to set the password complexity, instead of a hard-coding it.